### ISSUE TITLE
TypeScript: missing optional parameter conditions on can and cannot

### ISSUE THREAD

**Issue by ghiblin** [stalniy/casl#12]
in index.d.ts can and cannot function for AbilityBuilderParts miss optional condition parameter of type Object.
``` 
export abstract class AbilityBuilderParts {
  rules: Rule[]

  can(action: string | string[], subject: string | string[]): Rule

  cannot(action: string | string[], subject: string | string[]): Rule
}
```

**Comment by stalniy** [comment]
Thanks for the issue plan to fix in few hours!

**Comment by stalniy** [comment]
Fixed by a345da7 and published 1.0.5.

Thanks for using casl!

**Comment by stalniy** [comment]
Also fixes .d.ts for `rulesToQuery` and `rulesToQuery` in 1.0.6
